staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -61,7 +63,7 @@ const (
 func PatchResource(r rest.Patcher, scope *RequestScope, admit admission.Interface, patchTypes []string) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		// For performance tracking purposes.
-		trace := utiltrace.New("Patch", utiltrace.Field{Key: "url", Value: req.URL.Path}, utiltrace.Field{Key: "user-agent", Value: &lazyTruncatedUserAgent{req}}, utiltrace.Field{Key: "client", Value: &lazyClientIP{req}})
+		req, trace := requestWithTrace(req, "Patch", utiltrace.Field{Key: "url", Value: req.URL.Path}, utiltrace.Field{Key: "user-agent", Value: &internal.LazyTruncatedUserAgent{req}}, utiltrace.Field{Key: "client", Value: &internal.LazyClientIP{req}})
```

Comment:
also, aren't these fields duplicates from the main handler?

Reply:
They are. The catch is that they don't always both get included in the trace. The handler level trace output is only printed if the 30s threshold is exceeded, and the operation trace output (in this case "patch") is only printed if this operation handler exceeds 750ms. So when they both get output, there will be some redundant output, but when only one of them is printed, it is all needed.